superset/models.py

```diff
@@ -1527,8 +1527,8 @@ def fetch_metadata(self):
             if not any_date_col and dbcol.is_time:
                 any_date_col = col.name
 
-            quoted = "{}".format(
-                column(dbcol.column_name).compile(dialect=db.engine.dialect))
+            engine = self.database.get_sqla_engine()
+            quoted = col.compile(dialect=engine.dialect)
```

Comment:
^ The previous code created a column using a column name and compiled it, for some reason dialect override doesn't kick in. The new code takes existing column and compiles it using the dialect, that works.